### PR TITLE
python3Packages.skyfield: init at 1.42

### PR DESCRIPTION
**Pull request by vs49688**
###### Motivation for this change

Packaging skyfield and dependencies (assay, sgp4, and jplephem).

Can't run `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`, my system keeps OOM'ing due to an unrelated issue.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 161918` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.assay</li>
    <li>python310Packages.jplephem</li>
    <li>python310Packages.sgp4</li>
    <li>python310Packages.skyfield</li>
    <li>python39Packages.assay</li>
    <li>python39Packages.jplephem</li>
    <li>python39Packages.sgp4</li>
    <li>python39Packages.skyfield</li>
  </ul>
</details>

File: pkgs/development/python-modules/skyfield/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, fetchPypi, certifi, numpy, sgp4, jplephem
+, pandas, ipython, matplotlib, assay
+}:
```
Comment (jonringer) [inline pkgs/development/python-modules/skyfield/default.nix:3]: Please apply to the others
```suggestion
}:

```

File: pkgs/development/python-modules/skyfield/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  version = "1.42";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-NEf9Opqdq8IIDmpO+1bZiD3s8mGteObJs/GHxPx2Gs4=";
+  };
+
+  propagatedBuildInputs = [ certifi numpy sgp4 jplephem ];
+
+  checkInputs = [ pandas ipython matplotlib assay ];
```
Comment (jonringer) [inline pkgs/development/python-modules/skyfield/default.nix:15]: please add pytestCheckHook, as that will error if tests aren't ran.

sometimes the sdist have tests filtered out, so github should be pulled in that case

Comment (vs49688) [reply pkgs/development/python-modules/skyfield/default.nix:15]: Skyfield doesn't use pytest, it uses `assay`. I've changed checkPhase to use `assay --batch skyfield.tests`, but it's trying to pull down external files.

Should I pull them using `fetchurl`, or just disable the tests?

Comment (jonringer) [reply pkgs/development/python-modules/skyfield/default.nix:48]: I would just disable the network tests. We can hopefully trust upstream to publish only working releases, we want to run tests to ensure that it's not broken.

Comment (vs49688) [reply pkgs/development/python-modules/skyfield/default.nix:48]: I looked into that, it doesn't appear assay provides a way to disable them.

But, all the files are included in the Git repository, so I'll just change it to pull from that instead.

**Comment by vs49688** [comment]
ping on this?

**Comment by vs49688** [comment]
Ping again?

Something about this PR still causes nixpkgs-review to OOM:
```
Out of memory: Killed process 193171 (nix-env) total-vm:7482916kB, anon-rss:7340292kB, file-rss:228kB, shmem-rss:0kB, UID:1000 pgtables:14532kB oom_score_adj:200
```

**Comment by onny** [comment]
Result of `nixpkgs-review pr 161918` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.assay</li>
    <li>python310Packages.jplephem</li>
    <li>python310Packages.sgp4</li>
    <li>python310Packages.skyfield</li>
    <li>python39Packages.assay</li>
    <li>python39Packages.jplephem</li>
    <li>python39Packages.sgp4</li>
    <li>python39Packages.skyfield</li>
  </ul>
</details>

**Comment by onny** [comment]
Nice work! Could you try to add at least `pythonImportsCheck` to the packages if possible? Also explicitly adding `pytestCheckHook` to `checkInputs` is also good

**Comment by vs49688** [comment]
> Nice work! Could you try to add at least `pythonImportsCheck` to the packages if possible? Also explicitly adding `pytestCheckHook` to `checkInputs` is also good

I can add `pythonImportsCheck`, but not `pytestCheckHook` as none of the packages work with `pytest`, they use `assay` instead.

**Comment by onny** [comment]
Result of `nixpkgs-review pr 161918` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.assay</li>
    <li>python310Packages.jplephem</li>
    <li>python310Packages.sgp4</li>
    <li>python310Packages.skyfield</li>
    <li>python39Packages.assay</li>
    <li>python39Packages.jplephem</li>
    <li>python39Packages.sgp4</li>
    <li>python39Packages.skyfield</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

**Pull request merged**